mailbox/api/src/main/java/org/apache/james/mailbox/model/MailboxAnnotationKey.java

```diff
@@ -25,21 +25,18 @@
 
 import com.google.common.base.Objects;
 import com.google.common.base.Preconditions;
-import com.ibm.icu.text.UnicodeSet;
+import com.google.re2j.Pattern;
 
 public class MailboxAnnotationKey {
 
     public static final String SLASH_CHARACTER = "/";
     public static final String TWO_SLASH_CHARACTER = "//";
 
-    private static final UnicodeSet NAME_ANNOTATION_PATTERN = new UnicodeSet("[[a-z][A-Z][0-9]]")
-            .add(SLASH_CHARACTER)
-            .freeze();
+    private static final Pattern NAME_ANNOTATION_PATTERN = Pattern.compile("^([\\w\\d/]*)$");
```

Comment:
My preference goes to relying on Guava CharMatcher and avoid REGEX alltogether if that works.

Reply:
For instance:

```
    private static final CharMatcher NAME_ANNOTATION_PATTERN = CharMatcher.inRange('a', 'z')
        .or(CharMatcher.inRange('A', 'Z'))
        .or(CharMatcher.inRange('0', '9'))
        .or(CharMatcher.is('/'));
```

Passes the unit tests

Reply:
Switched to CharMatcher